ckb-bin/src/subcommand/init.rs

```diff
@@ -35,9 +35,11 @@ pub fn init(args: InitArgs) -> Result<(), ExitCode> {
             format!(
                 "[block_assembler]\n\
                  code_hash = \"{}\"\n\
-                 args = [ \"{}\" ]",
+                 args = [ \"{}\" ]\n\
+                 data = \"{}\"",
                 hash,
-                args.block_assembler_args.join("\", \"")
+                args.block_assembler_args.join("\", \""),
+                args.block_assembler_data.unwrap_or_default()
```

Comment:
Default does not work, it is not a valid hex string. Either omit data or use "0x" as default.